### ISSUE TITLE
`develop` branch diverges from current `master` snapshot after #23 merge

### ISSUE THREAD

**Issue by mhhansen** [payway-ar/plugin-prestashop#26]
### Preconditions (*)

1. `develop` branch has been merged directly on top of `master` branch (unconfirmed)
2.

### Steps to reproduce (*)

1. Go to the _Pull requests_ tab and create a new PR
2. Select `develop` as a base branch, and compare with `master` (or the other way around)

### Expected result (*)

1. No diff should be shown related to #23, or any other piece of code that has already been merged into both branches
2.

### Actual result (*)

1. Differences in two files are displayed, causing the WIP #25 to not be trusted to be merged, cause drags changes outside of it's scope:
    * decidir/controllers/front/payment.php - ([view it within PR#25](https://github.com/decidir/dec_prestashop/pull/25/files?authenticity_token=tJm6L1cJr8kRHvb2IzuW02bPC75l0osJVrmKLaUgj2lElsvml%2BE7%2BGeq22hQ6bZhVFXmWrlT2RUT1BAQqXQcLg%3D%3D&file-filters%5B%5D=.jpg&file-filters%5B%5D=.json&file-filters%5B%5D=.md&file-filters%5B%5D=.php&file-filters%5B%5D=.png#diff-0c3cc06c33e0172a3e9fb5567bd38338d7e418c6ca9e16877f9fde8d6c2bf067L81))
    * decidir/includes/cbs-retail.php - ([view it within PR#25](https://github.com/decidir/dec_prestashop/pull/25/files?authenticity_token=xFR3pi2SVuzieLfb2ktUrkxqVr0YnY%2Fjdt12Gm3wyBY0WwZv7XrC3ZTMmkWpmXQcfvC7WcQc3f8zsOwnYaRbUQ%3D%3D&file-filters%5B%5D=.jpg&file-filters%5B%5D=.json&file-filters%5B%5D=.md&file-filters%5B%5D=.php&file-filters%5B%5D=.png#diff-436045c38c7df7582572f515420cea971f22fb8a788e72d22d059e8f12276d84))
2.

---
### Additional Information
This will be blocking progress of #25 (once it changes the WIP status).

I've spotted this while working on #25 (which started from `master`), but now I've the `cbs-retail.php` file (among other commits):
[decidir/includes/cbs-retail.php](https://github.com/decidir/dec_prestashop/pull/25/files#diff-436045c38c7df7582572f515420cea971f22fb8a788e72d22d059e8f12276d84)

Can't precise exactly what I've caused this, it may be the commit hash, but if that is the case, shouldn't be a diff in the code base
Here are the two PRs related to this issue:
* Original PR against `develop`: #23
* Merge `develop` into `master`: #24

Here is a comparison (of one of the diffs within the current state of #25), for the `cbs-retail.php` file:
* `develop` branch: https://github.com/decidir/dec_prestashop/blob/develop/decidir/includes/cbs-retail.php
* `master` branch: https://github.com/decidir/dec_prestashop/blob/master/decidir/includes/cbs-retail.php

As you can see, there are no diff between both versions, but still the WIP PR shows it.
 
**room for a personal thought:** I believe, as a rule of thumb, the original _bugfix/feature/etc_ branch, shouldn't be closed along with the merge action against `develop`. 
Instead, it should remain on remote and, once it's OK to be merged into `master`, a new PR between the _bugfix/feature/etc_ branch against `master` branch needs to be created. Not only because it contains the exact commits hash and history, but when multiple developers are working against `develop`, merging `develop` directly on top of `master` branch, can cause unexpected results.

Hope this helps!
